Utilities/Install-CustomScript.ps1

```diff
@@ -0,0 +1,177 @@
+##############################################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# Install-CustomScript.ps1
+<#
+.SYNOPSIS
+
+.PARAMETER
+	-AzureSecretsFile, the path of Azure secrets file
+	-VmName, the name of the VM to install the CustomScript extension, if no VmName is provided, it will be installed on all VMs
+	-ResourceGroupName, the resource group name of the VMs to install the CustomScript extension, if no ResourceGroupName is provided, it will be installed on all VMs in the subscription
+	-FileUris, comma separated Uris of the custom scripts
+	-CommandToRun, the command to run on the VM
+	-StorageAccountName, the name of the storage account that contains the custom scripts
+	-StorageAccountKey, the key of the storage account that contains the custom scripts
+	-OSType, the type of the OS, Linux or Windows, valid only if neither VmName nor ResourceGroupName is provided
+
+.NOTES
+	Creation Date:
+	Purpose/Change:
+
+.EXAMPLE
+	.\Utilities\Install-CustomScript.ps1 -AzureSecretsFile <PathToSecretFile> -VmName <VmName> -ResourceGroupName <RgName> `
+		-FileUris "https://teststorageaccout.blob.core.windows.net/script/install.sh" -CommandToRun "bash install.sh" `
+		-StorageAccountName <StorageAccoutName> -StorageAccountKey <StorageAccountKey> -OSType Linux
+#>
+###############################################################################################
+
+param
+(
+	[Parameter(Mandatory=$true)]
+	[String] $AzureSecretsFile,
+	[String] $VmName = "",
+	[String] $ResourceGroupName = "",
+	[Parameter(Mandatory=$true)]
+	[String] $FileUris,
+	[Parameter(Mandatory=$true)]
+	[String] $CommandToRun,
+	[String] $StorageAccountName = "",
+	[String] $StorageAccountKey = "",
+	[ValidateSet('Linux','Windows', IgnoreCase = $false)]
+	[String] $OSType = "Linux"
```

Comment:
no WSL?

Reply:
It's the "Operation System" field of the VM displayed in Azure portal. Now it contains only "Linux" and "Windows"